### PR TITLE
Improve Youtube Music provider

### PR DESCRIPTION
**Pull request by MarvinSchenkel**
This PR adds:
- A brand new parser, based on the ytmusicapi package -> Fixes parser errors and improves overall stability
- Fixes max limit of 25 objects
- Grabs highest quality image by default as a thumbnail

### CODE REVIEW

File: music_assistant/music_providers/ytmusic/ytmusic.py [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+        thumbnails_sorted = sorted(
+            thumbnails_obj, key=itemgetter("width"), reverse=True
+        )
+        return [
+            MediaItemImage(ImageType.THUMB, thumb["url"]) for thumb in thumbnails_sorted
+        ]
+
+    @classmethod
+    async def _parse_stream_format(cls, track_obj: dict) -> dict:
+        """Grab the highes available audio stream from available streams."""
```
Comment (marcelveldt) [inline music_assistant/music_providers/ytmusic/ytmusic.py:466]:
```suggestion
        """Grab the highest available audio stream from available streams."""
```

File: music_assistant/music_providers/ytmusic/ytmusic.py [hunk 2]
```diff
@@ -0,0 +1,495 @@
...
+        signature = await self._decipher_signature(
+            ciphered_signature=cipher_parts["s"], item_id=item_id
+        )
+        url = cipher_parts["url"] + "&sig=" + signature
+        return url
+
+    @classmethod
+    async def _parse_thumbnails(cls, thumbnails_obj: dict) -> List[MediaItemImage]:
+        """Parse and sort a list of thumbnails from highest quality to lowest quality."""
+        thumbnails_sorted = sorted(
```
Comment (marcelveldt) [inline music_assistant/music_providers/ytmusic/ytmusic.py:457]: If the thumbnails are the same there is no need to store them all. In that case only store the highest quality as thumbnail creation is all handled globally

Comment (MarvinSchenkel) [reply music_assistant/music_providers/ytmusic/ytmusic.py:457]: They only differ in size, which might be handy for the front-end. Or does the front-end create small version automatically?

Comment (MarvinSchenkel) [reply music_assistant/music_providers/ytmusic/ytmusic.py:457]: Done

**Pull request merged**